pyquil/api/_abstract_compiler.py

```diff
@@ -78,6 +78,7 @@ def __init__(
         quantum_processor: AbstractQuantumProcessor,
         timeout: float,
         client_configuration: Optional[QCSClient] = None,
+        quilc_client: Optional[RPCQClient] = None,
```

Comment:
I wonder if we could export a `QuilcClient` class from qcs-sdk-python that would be based off of a Rust enum. We'd have a variant for each client implementation and constructors for users to build an instance of that enum with whichever implementation they wanted. We could use that as the type here without worrying about expanding it.

Reply:
^ this sounds great